v3/src/components/graph/dotplotdots.tsx

```diff
@@ -1,122 +1,114 @@
+import {
+  plotProps,
+  transitionDuration,
+  InternalizedData,
+  defaultRadius,
+  defaultDiameter,
+  dragRadius
+} from "./graphing-types"
+import {useDragHandlers, useSelection} from "./graph-hooks/graph-hooks"
+import {IDataSet} from "../../data-model/data-set"
+import {observer} from "mobx-react-lite"
+import {getScreenCoord} from "./graph-utils/graph_utils"
+
+
+export const DotPlotDots = memo(observer(function DotPlotDots(props: {
   dots: plotProps,
   plotWidth: number,
   plotHeight: number,
   xMin: number,
   xMax: number,
-  data: worldData[],
-  setData: React.Dispatch<React.SetStateAction<worldData[]>>
-  setHighlightCounter: React.Dispatch<React.SetStateAction<number>>
+  worldDataRef:  React.MutableRefObject<IDataSet | undefined>,
+  graphDataRef: React.MutableRefObject<InternalizedData>,
   dotsRef: React.RefObject<SVGSVGElement>
 }) {
   const {
-      data, setData, dotsRef, plotWidth, plotHeight, setHighlightCounter, xMax, xMin,
+      worldDataRef, graphDataRef, dotsRef, plotWidth, plotHeight, xMax, xMin,
       dots: {xScale, yScale}
     } = props,
-    [dragID, setDragID] = useState(-1),
+    [dragID, setDragID] = useState(''),
     [refreshCounter, setRefreshCounter] = useState(0),
     currPos = useRef({x: 0}),
     target = useRef<any>(),
     [firstTime, setFirstTime] = useState<boolean | null>(true),
-    selectedDataObjects = useRef<{ [index: number]: { x: number } }>({}),
-    [forceRefreshCounter, setForceRefreshCounter]=useState(0)
+    selectedDataObjects = useRef<{ [index: string]: { x: number } }>({}),
+    [forceRefreshCounter, setForceRefreshCounter] = useState(0)
 
   const onDragStart = useCallback((event: MouseEvent) => {
 
-      const selectPoint = (iData: worldData[]) => {
-        iData.forEach((datum) => {
-          if (datum.id === tItsID && !datum.selected) {
-            datum.selected = true
-            setHighlightCounter(prevCounter => ++prevCounter)
-          }
-        })
-        return iData
-      }
-
       if (firstTime) {
         setFirstTime(false) // We never want to animate points on drag
       }
-      // target.current = event.target as SVGSVGElement
+      const xAttrID = graphDataRef.current.xAttributeID
       target.current = select(event.target as SVGSVGElement)
-      const tItsID = Number(target.current.property('id'))
+      const tItsID: string = target.current.property('id')
       if (target.current.node()?.nodeName === 'circle') {
         target.current.transition()
           .attr('r', dragRadius)
         setDragID(() => tItsID)
         currPos.current = {x: event.clientX}
-        setData(selectPoint(data))
+        worldDataRef.current?.selectCases([tItsID])
+        // Record the current values so we can change them during the drag and restore them when done
+        worldDataRef.current?.selection.forEach(anID => {
+          selectedDataObjects.current[anID] = {
+            x: worldDataRef.current?.getNumeric(anID, xAttrID) ?? 0
```

Comment:
It probably shouldn't ever happen, but I would think that defaulting to `NaN` would make more sense than `0`. Otherwise, it would run the risk of being plotted as data.